src/raven.js

```diff
@@ -843,6 +843,38 @@ function makeRequest(opts) {
     img.src = src;
 }
 
+function makeXhrRequest(opts) {
+    var request = new XMLHttpRequest();
+    request.onreadystatechange = function (e) {
+        if (request.readyState !== 4) {
+            return;
+        }
+
+        if (request.status === 200) {
+            if (opts.onSuccess) {
+                opts.onSuccess();
+            }
+        } else {
+            if (opts.onError) {
+                opts.onError();
+            }
+        }
+    };
+
+    // NOTE: auth is intentionally sent as part of query string (NOT as custom
+    //       HTTP header) so as to avoid preflight CORS requests
+    request.open('POST', opts.url + '?' + urlencode(opts.auth));
+    request.send(JSON.stringify(opts.data));
+}
+
+function makeRequest(opts) {
+    var hasCORS =
+        'withCredentials' in new XMLHttpRequest() ||
+        typeof XDomainRequest !== 'undefined';
```

Comment:
> Realistically – yes, unless someone has removed it (e.g. the host page or a browser extension)

What about the IE case though since they use XDomainRequest? Or do I misunderstand this?